### PR TITLE
New image builder without molecule using podman

### PR DESCRIPTION
**Pull request by t-woerner**
The new image builder is not using molecule and uses podman directly for the generation of the ansible-test images.

Two additional services are installed to simplify the use of the container in the test:

- fixnet.service uses /root/fixnet.sh to fix IP address of the server in /etc/hosts and to set localhost as the nameserver. This service is executed before IPA is started. This eliminates the need to restart the IPA server after the container has been started and the IPs have been fixed.
- fixipaip.service uses /root/fixipaip.sh to fix the IP address of the IPA dnsrecords of server and ipa-ca.

With these services it is now only needed to wait till all services in the container are started. There is no need to restart the IPA server anymore. Simply use something like this before starting the tests:

    while [ -n "$(podman exec ansible-test systemctl list-jobs | grep -vi 'no jobs running')" ]; do echo "waiting.."; sleep 5; done

New files
- infra/image/build.sh
- infra/image/dockerfile/c8s
- infra/image/dockerfile/c9s
- infra/image/dockerfile/fedora-latest
- infra/image/dockerfile/fedora-rawhide
- infra/image/inventory
- infra/image/system-service/fixipaip.service
- infra/image/system-service/fixipaip.sh
- infra/image/system-service/fixnet.service
- infra/image/system-service/fixnet.sh

### CODE REVIEW

File: infra/image/build.sh [hunk 1]
```diff
@@ -0,0 +1,66 @@
+#!/bin/bash -eu
```
Comment (f-trivino) [inline infra/image/build.sh:1]: really minor: what about "#!/bin/bash -euo pipefail" it ensures the script exits on errors, the unset vars as errors, piped commands...

File: infra/image/build.sh [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
```
Comment (f-trivino) [inline infra/image/build.sh:2]: not for automated scenario, but I think this script could be also used locally, what about do some additional checks?

# Ensure podman and ansible-playbook commands are available
command -v podman >/dev/null 2>&1 || { echo "ERROR: podman is not installed or not in PATH"; exit 1; }
command -v ansible-playbook >/dev/null 2>&1 || { echo "ERROR: ansible-playbook is not installed or not in PATH"; exit 1; }

File: infra/image/dockerfile/c10s [hunk 3]
```diff
@@ -0,0 +1,25 @@
+FROM quay.io/centos/centos:stream10-development
+ENV container=podman
+
+RUN rm -fv /var/cache/dnf/metadata_lock.pid; \
```
Comment (f-trivino) [inline infra/image/dockerfile/c10s:4]: uhmm,  I would combine commands using "&&" to ensure they run in a single layer and handle errors properly

what about:

```
# Install necessary packages and clean up
RUN rm -fv /var/cache/dnf/metadata_lock.pid && \
    dnf makecache && \
    dnf --assumeyes install \
        /usr/bin/python3 \
        /usr/bin/dnf-3 \
        sudo \
        bash \
        systemd \
        procps-ng \
        iproute && \
    rm -rf /var/cache/dnf/
```

File: infra/image/system-service/fixipaip.sh [hunk 4]
```diff
@@ -0,0 +1,13 @@
+#!/bin/bash -eu
+
+HOSTNAME=$(hostname)
+IP=$(hostname -I | cut -d " " -f 1)
+
```
Comment (f-trivino) [inline infra/image/system-service/fixipaip.sh:5]: should we double check if we have hostname properly configured inside the container?

if [ -z "${HOSTNAME}" ] || [ -z "${IP}" ]; then
    echo "ERROR: Failed to retrieve hostname or IP address"
    exit 1
fi

File: infra/image/system-service/fixnet.service [hunk 5]
```diff
@@ -0,0 +1,12 @@
+[Unit]
+Description=Fix server IP in IPA Server
+Wants=network.target
+After=network.target
+Before=ipa.service
+
+[Service]
+Type=oneshot
+ExecStart=/root/fixnet.sh
+
```
Comment (f-trivino) [inline infra/image/system-service/fixnet.service:10]: you could add:

StandardOutput=journal
StandardError=journal

if you want to capture the scripts errors in the journal logs.

File: infra/image/build.sh [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+TOPDIR="$(readlink -f "${BASEDIR}/../..")"
+
+scenario=${1:-}
+name="ansible-test"
+hostname="ipaserver.test.local"
+cpus="2"
+memory="4g"
+quayname="quay.io/ansible-freeipa/upstream-tests"
+
+if [ -z "${scenario}" ]; then
```
Comment (f-trivino) [inline infra/image/build.sh]: where is the valid list of scenarios? should we add a comment?

Comment (f-trivino) [reply infra/image/build.sh]: perhaps we could printout
```
ls -1 ./dockerfile/
```

File: utils/shfun [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+# This file is meant to be source'd by shell scripts
+
+SCRIPTDIR="$(dirname -- "$(readlink -f "${BASH_SOURCE[0]}")")"
+
+. "${SCRIPTDIR}/shlog"
+
+[ -n "$(command -v python3)" ] && python="$(command -v python3)" || python="$(command -v python2)"
+export python
+
+trap interrupt_exception SIGINT
```
Comment (f-trivino) [inline utils/shfun:11]: cosmetic: should we move the definition of interrupt_exception before it is used?

File: infra/image/build.sh (hunk 2 above)
Comment (t-woerner) [reply infra/image/build.sh:2]: The check for ansible-playbook is already there. I added the one for podman with the latest commit.

File: infra/image/system-service/fixipaip.sh [hunk 8]
```diff
@@ -0,0 +1,13 @@
...
+echo "SomeADMINpassword" | kinit -c ansible_freeipa_cache admin
```
[the remark this answers is not in the document]
Comment (t-woerner) [reply infra/image/system-service/fixipaip.sh:6]: I will fix the script according to https://www.shellcheck.net/wiki/SC2181

**Review by f-trivino — APPROVED** [review]
LGTM!, 
I tested infra/ scripts , they work for me on my local env.

File: infra/image/dockerfile/c10s [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+    /usr/bin/python3 \
+    /usr/bin/dnf-3 \
+    sudo \
+    bash \
+    systemd \
+    procps-ng \
+    iproute; \
+rm -rf /var/cache/dnf/;
+
+COPY system-service/fixnet.sh /root/
```
[the remark this answers is not in the document]
Comment (t-woerner) [reply infra/image/dockerfile/c10s:16]: Added to the first commit for all dockerfiles.

**Pull request merged**